include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -141,6 +141,18 @@ class LAPACKFullMatrix : public TransposeTable<number>
   void add (const number                    a,
             const LAPACKFullMatrix<number> &A);
 
+  /**
+   * Perform a rank-1 update of a symmetric matrix
+   * $ A \leftarrow A + a \, \rm v \rm v^T $.
+   *
+   * This function also works for Cholesky factorization. Updating ($a>0$) is
+   * performed via Givens rotations, whereas downdating ($a<0$) via hyperbolic
+   * rotations. Note that the later case might lead to a negative definite
+   * matrix in which case the error will be thrown.
+   */
+  void add(const number a,
+           const Vector<number> &v);
```

Comment:
I would give this a more verbose name, say `rank1_update`. We already use `add` for too many other things. Or do we already use it somewhere?

Reply:
no, we don't have anywhere `A = A + v v`. I am fine with renaming it to `rank1_update` if you find it more descriptive.